server/api/search.js

```diff
@@ -0,0 +1,139 @@
+import { Router } from 'express';
+import { Base64 } from 'js-base64';
+
+import sequelize from '../models';
+import { ValidationError } from '../models/validator';
+import { getPageInfo, isNumber } from '../util';
+import { RESULT_PER_PAGE } from '../../constant';
+
+const router = Router();
+const { Op } = sequelize.Sequelize;
+
+export const parseQuery = (q) => {
+  const keywords = [];
+  const tags = [];
+  const users = [];
+  q.split(' ').forEach((word) => {
+    if (word[0] === '#') {
+      tags.push(`%${word.slice(1)}%`);
+    } else if (word[0] === '@') {
+      users.push(`%${word.slice(1)}%`);
+    } else {
+      keywords.push(`%${word}%`);
+    }
+  });
+  return [keywords, tags, users];
+};
+
+router.get('/search', async (req, res, next) => {
+  const {
+    headers: { host },
+    protocol,
+    query,
+  } = req;
+  let offset = 0;
+  let q = decodeURIComponent(query.q || '');
```

Comment:
I suggest the whole pagination logic to in an separate testable module in stead of putting inline here in the controller.

i.e. in `server/utils/index:getPageInfo` and the `parseQuery` here belong to `server/utils/paginator` and create `Paginate` class and `PageInfo` class to represent the logic here.